discovery-infra/test_infra/tools/terraform_utils.py

```diff
@@ -18,15 +21,35 @@ def __init__(self, working_dir: str):
         self.init_tf()
 
     def init_tf(self) -> None:
-        self.tf.cmd("init -plugin-dir=/root/.terraform.d/plugins/", raise_on_error=True)
+        self.tf.cmd("init", raise_on_error=True)
+
+    def select_defined_variables(self, **kwargs):
+        supported_variables = self.get_variable_list()
+        return {k: v for k, v in kwargs.items() if v and k in supported_variables}
+
+    def get_variable_list(self):
+        results = list()
+
+        for tf_file in pathlib.Path(self.working_dir).glob("*.tf"):
+            with open(tf_file, "r") as fp:
+                terraform_file_dict = hcl2.load(fp)
+                results += terraform_file_dict["variable"] if "variable" in terraform_file_dict else list()
+
+        return list(map(lambda d: next(iter(d)), results))
 
     def apply(self, refresh: bool = True) -> None:
-        return_value, output, err = self.tf.apply(no_color=IsFlagged, refresh=refresh, input=False, skip_plan=True)
+        return_value, output, err = self.tf.apply(no_color=IsFlagged, refresh=refresh,
+                                                  input=False, skip_plan=True)
         if return_value != 0:
             message = f"Terraform apply failed with return value {return_value}, output {output} , error {err}"
             logging.error(message)
             raise Exception(message)
 
+    def set_and_apply(self, refresh: bool = True, **kwargs) -> None:
+        defined_variables = self.select_defined_variables(**kwargs)
+        self.change_variables(defined_variables)
+        self.apply(refresh=refresh)
```

Comment:
How those modifications are related to VSphere?

Reply:
This code allow us to take the configuration objects, select only the relevant terraform properties and then apply.
It's used by the controller but basically it's a utility method.
I'm trying to splitting out things